src/cmd-build

```diff
@@ -259,13 +264,26 @@ prepare_git_artifacts "${configdir_gitrepo}" "${PWD}/coreos-assembler-config-git
 
 extra_compose_args=()
 
+# Apply autolock from another build for this version if no base lockfile
+# exists. Do this before so that overrides come after.
+if [ ! -f "${manifest_lock}" ] && [ -n "${VERSION}" ] && [ -n "${AUTOLOCK}" ]; then
```

Comment:
i'm starting to think that we should be checking these things up front rather than here.

i.e. when parsing arguments:

```
if --autolock was passed:
    error if -f "${manifest_lock}"
    error if -z "${VERSION}"
```

Then this could would just check `if [ -n "${AUTOLOCK}" ]` and do the work.

Reply:
I guess another way to say this is. Can we be more strict here and error if someone passed `--autolock` in a scenario that doesn't make sense?

Reply:
`--autolock` is designed to nicely no-op if a base lockfile is provided so that we don't have to add conditionals in the pipeline. We could make it hard require `--version` though.

Reply:
I kind of feel like this is something the user shouldn't ask for if they don't want it. i.e. I think maybe conditionals in the pipeline would be appropriate.

Maybe... I'm still exploring my feelings on it.

Reply:
Heh, that's what the "auto" part hints at; it automatically creates a lockfile if necessary. I actually initially debated not making this an option at all and just having it do it automatically if it detects you're trying to "complete" an existing multi-arch build, because I think you almost always want this in that case. But it's easier to experiment with to start if it's opt-in.

Reply:
> I actually initially debated not making this an option at all and just having it do it automatically if it detects you're trying to "complete" an existing multi-arch build, because I think you almost always want this in that case.

Agree. Let's just make it automatic without an option (which is what you've done in the latest upload).